### PR TITLE
Asset transfer private-data go-chaincode - Add unit tests

### PR DESCRIPTION
**Pull request by sijocherian**
- refactored to chaincode package as best practice for organization
- Added unit tests for query & transfer asset chaincode functions, with counterfeiter mocks

### CODE REVIEW

**Comment by nikhil550** [comment]
executed successfully locally

**Comment by sijocherian** [comment]
> Sorry, looks like you'll need to rebase before we merge.

Done. Had some issues with rebase, resolved using another intermediate branch.

File: asset-transfer-private-data/application-javascript/app.js [hunk 1]
```diff
@@ -221,11 +234,23 @@ async function main() {
...
             let dataForDelete = { assetID: assetID2 };
+            try {
+                //Non-owner Org2 should not be able to DeleteAsset. Expect an error from DeleteAsset
+                console.log('--> Attempt Transaction: as Org2 DeleteAsset ' + assetID2);
+                statefulTxn = contractOrg2.createTransaction('DeleteAsset');
+                tmapData = Buffer.from(JSON.stringify(dataForDelete));
+                statefulTxn.setTransient({
+                    asset_delete: tmapData
+                });
+                result = await statefulTxn.submit();
```
Comment (sijocherian) [inline asset-transfer-private-data/application-javascript/app.js:246]: beyond the change in #309 , added this call to Delete, where error is expected and corresponding check in the chaincode

**Comment by denyeart** [comment]
Looks good now, but still has conflicts so I can't yet merge...

**Comment by sijocherian** [comment]
github was not showing conflict. Trying to just trigger the github state by changing target branch and resetting back

**Pull request merged**